### ISSUE TITLE
General system settings function

### ISSUE THREAD

**Issue by ThomDietrich** [ThomDietrich/openhabian#36]
Add one more function which is executed at the end of the initial setup and configures a few details like

```
adduser openhab dialout
adduser openhab tty
adduser pi dialout
adduser pi tty
```

Besides, this needs to go inside Homematic function:

```
adduser pi homegear
```

**Comment by BClark09** [comment]
```
setcap 'cap_net_raw,cap_net_admin=+eip cap_net_bind_service=+ep' `realpath /usr/bin/java`
```

For easy use of network and amazon dash bindings?

**Comment by ThomDietrich** [comment]
@BClark09 Good idea. I've actally added this information to the network binding readme: http://docs.openhab.org/addons/bindings/network/readme.html#permissions

Do you need the additional permissions for Dash?

**Comment by ThomDietrich** [comment]
Btw: https://github.com/openhab/openhab-docs/pull/135

**Comment by BClark09** [comment]
That's right, setcap also overwrites anything previously set to Java, so it's best to do all three to cover all (current bases)

**Comment by ThomDietrich** [comment]
http://docs.openhab.org/installation/linux.html#recommended-additional-setup-steps

**Comment by binderth** [comment]
I'm not sure here. In my opinion either openHABian pre-configures the system for *all* bindings - or (that's what I tend to prefer), each binding should provide detailed information on those configurations.

e.g. I use the comfoair-binding. In my case, I use a 10€ USB2RS232-converter. But you could also attach it to the GPIO-ports... How can openHABian support all bindings and their respective configuration? In my opinion the goal should be to provide Information on how to configure your system in the binding-documentation - of course this would be platform independent so you need to describe basic Linux, ARM and Windows configuration for each binding, which relies on the underlying system... But in my experience with openHAB: This is the main breakpoint. I'm not the heavy user and I'm not the first time Pi user - I'm just in the middle an experienced Linux-User - and sometimes I need to take a few extra rounds to get it sorted out on where and how to configure stuff... ;) I wished, I had some more system-related information in binding documentations.

**Comment by ThomDietrich** [comment]
Hello @binderth, I'm not sure if I understood you right.
openHABian's goal is to simplify the installation of openHAB and the components useful around it. openHABian will not interact, configure or modify any openHAB bindings.

**Comment by ThomDietrich** [comment]
@BClark09 I've just implemented the first two postings.

**Comment by ThomDietrich** [comment]
https://github.com/ThomDietrich/openhabian/commit/62804c4301444ee2ee47d274430d1e9850049ad9

**Comment by binderth** [comment]
@ThomDietrich then I got it wrong in the first place. I refered to pre-configuring the Linux for the use with bindings, e.g. your post on Nov 10th. In my eyes, this is some configuration, which should be at the respected bindings (here the configuration of ports). I understand, your suggestion is very low level and is perhaps useful for absolute beginners. They could plug in some USB and the openhab user can use the ports. In my experience in projects, the scope widens very fast and you're into configuring special "escimo cases" - like amazon dash buttons. (no hard feelings, though! 😄).

Long Story short: configuration also of the system for use with specific bindings belong to the description of the respected binding-docu. I agree, that's not the case as of today...

**Comment by ThomDietrich** [comment]
While I get your argument and can support it, I don't think that's the case here. Adding the openHAB user to the dialout group or setting java network permissions is something needed by multiple bindings while not making any (considerable) harm to your system. That's why I would deem these changes acceptable as a general setting in openHABian. There are other cases, where special settings are only applied after selecting a openhabian-config menu entry.

Yes, settings needed for specific bindings should be described in the documentation of these bindings. That's how it is ;) Check [here](http://docs.openhab.org/installation/linux.html#recommended-additional-setup-steps) or [here](http://docs.openhab.org/addons/bindings/network/readme.html#permissions).

**Comment by binderth** [comment]
we're talking the same, fine with me! ;) 💯 
just wanted to keep you from overscoping! 😀

**Comment by ThomDietrich** [comment]
:) Thanks.
What's your use case and experience? Would you like to contribute something to openHABian?

**Comment by binderth** [comment]
I'm not the LINUX-Systems guy, and also not the JAVA-programmer guy. So my usecase is to get some start (which openHABian is very good at). I'd like to see more documentation and HowTos (that's more to openHAB, than openHABian, I guess); so, that's what I could offer!

**Comment by BClark09** [comment]
@ThomDietrich, Done but I've kept them as headings for now. Just because Linux needs a little bit extra information.
